### ISSUE TITLE
Failure to install with Python 3.9, probably because of numba/llvmlite

### ISSUE THREAD

**Issue by warrickball** [lightkurve/lightkurve#951]
#### Problem description

I just tried to upgrade from 1.11.3 to 2.0.0 on Fedora 33 & Python 3.9.1 using `pip` and the installation eventually fails because of `llvmlite` via `numba`. I recently hit the same error when trying to install `numba`, so this issue is upstream. But is there a way to work around it until the relevant packages become available on Python 3.9? 

For anyone interested, I believe these are relevant issues for [llvmlite](https://github.com/numba/llvmlite/issues/669) and [numba](https://github.com/numba/numba/issues/6345). Here's the full error message:
````
Building wheels for collected packages: llvmlite
  Building wheel for llvmlite (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-43form_n/llvmlite/setup.py'"'"'; __file__='"'"'/tmp/pip-install-43form_n/llvmlite/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-ce4tbn_t
       cwd: /tmp/pip-install-43form_n/llvmlite/
  Complete output (15 lines):
  running bdist_wheel
  /usr/bin/python3 /tmp/pip-install-43form_n/llvmlite/ffi/build.py
  LLVM version... 11.0.0
  
  Traceback (most recent call last):
    File "/tmp/pip-install-43form_n/llvmlite/ffi/build.py", line 191, in <module>
      main()
    File "/tmp/pip-install-43form_n/llvmlite/ffi/build.py", line 181, in main
      main_posix('linux', '.so')
    File "/tmp/pip-install-43form_n/llvmlite/ffi/build.py", line 143, in main_posix
      raise RuntimeError(msg)
  RuntimeError: Building llvmlite requires LLVM 10.0.x or 9.0.x, got '11.0.0'. Be sure to set LLVM_CONFIG to the right executable path.
  Read the documentation at http://llvmlite.pydata.org/ for more information about building llvmlite.
  
  error: command '/usr/bin/python3' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for llvmlite
  Running setup.py clean for llvmlite
Failed to build llvmlite
DEPRECATION: Could not build wheels for llvmlite which do not use PEP 517. pip will fall back to legacy 'setup.py install' for these. pip 21.0 will remove support for this functionality. A possible replacement is to fix the wheel build issue reported above. You can find discussion regarding this at https://github.com/pypa/pip/issues/8368.
Installing collected packages: certifi, requests, memoization, llvmlite, numba, lightkurve
    Running setup.py install for llvmlite ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-43form_n/llvmlite/setup.py'"'"'; __file__='"'"'/tmp/pip-install-43form_n/llvmlite/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-e07m0q5x/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/wball/.local/include/python3.9/llvmlite
         cwd: /tmp/pip-install-43form_n/llvmlite/
    Complete output (18 lines):
    running install
    running build
    got version from file /tmp/pip-install-43form_n/llvmlite/llvmlite/_version.py {'version': '0.34.0', 'full': 'c5889c9e98c6b19d5d85ebdd982d64a03931f8e2'}
    running build_ext
    /usr/bin/python3 /tmp/pip-install-43form_n/llvmlite/ffi/build.py
    LLVM version... 11.0.0
    
    Traceback (most recent call last):
      File "/tmp/pip-install-43form_n/llvmlite/ffi/build.py", line 191, in <module>
        main()
      File "/tmp/pip-install-43form_n/llvmlite/ffi/build.py", line 181, in main
        main_posix('linux', '.so')
      File "/tmp/pip-install-43form_n/llvmlite/ffi/build.py", line 143, in main_posix
        raise RuntimeError(msg)
    RuntimeError: Building llvmlite requires LLVM 10.0.x or 9.0.x, got '11.0.0'. Be sure to set LLVM_CONFIG to the right executable path.
    Read the documentation at http://llvmlite.pydata.org/ for more information about building llvmlite.
    
    error: command '/usr/bin/python3' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-43form_n/llvmlite/setup.py'"'"'; __file__='"'"'/tmp/pip-install-43form_n/llvmlite/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-e07m0q5x/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/wball/.local/include/python3.9/llvmlite Check the logs for full command output.
````

#### Environment

* Fedora 33
* Python 3.9.1
* Lightkurve 2.0.0
* Installation via `pip`

**Comment by barentsen** [comment]
Thanks @warrickball.  This is useful feedback.

My understanding from https://github.com/numba/numba/issues/6345 is that this should be resolved with numba 0.53, which hasn't been released yet.  Version "0.53.0rc1.post1" is available on pypi however.  Perhaps this should be a minimum dependency for Lightkurve.

Alternatively, we could look into making numba an optional dependency, because only Lightkurve's noise correction tools need it.

**Comment by warrickball** [comment]
Yeah, I've just been reading further and from what I can see, this might be resolved quite soon. I presume Python versions don't usually change the language at the level that has affected numba this time, so I'm happy to wait and see before I think there's a good reason for the lightkurve team to commit time to this. Partly out of curiosity, I might try installing the numba and llvmlite release candidates and see how lightkurve reacts...

**Comment by warrickball** [comment]
In case anyone else runs into this, I've managed to install lightkurve by first installing the numba 0.53 release candidate with
```
python3 -m pip install -U numba==0.53.0rc1.post1 --user
```
and just successfully went through the [quickstart](http://docs.lightkurve.org/quickstart.html). That's good enough for me for now so I'm happy for anyone to close this (short-lived) issue.

If I hit any numba-related issues, I'll wait until 0.53 is fully released before raising issues here.

**Comment by barentsen** [comment]
Thanks for running this experiment @warrickball! :+1:

I went ahead and added `numba>=0.53.0rc1.post1` as a minimum dependency. I verified that this appears to resolve the issue, both on my system and on GitHub Actions, and released Lightkurve v2.0.1 to include this change.

Thanks again for drawing my attention to this! :+1:
